doc/mgmt/OpenConfig_Platform.md

```diff
@@ -0,0 +1,454 @@
+
+
+# Scope
+- This document describes the high level design of OpenConfig telemetry for states of transceiver components via gNMI in SONiC.
+- This does not cover the SONiC KLISH CLI.
+- This covers openconfig-platform-transceiver.yang and its sub-modules.
+- This does not cover gNMI SUBSCIRBE STREAM ON_CHANGE and TARGET_DEFINED.
+- Supported attributes in OpenConfig YANG tree:
+
+```  
+module: openconfig-platform
+  +--rw components
+     +--rw component* [name]
+        +--ro state
+        |  +--ro serial-no?                             string
+        +--rw oc-transceiver:transceiver
+           +--ro oc-transceiver:state
+           |  +--ro oc-transceiver:connector-type?      identityref
+           |  +--ro oc-transceiver:vendor?              string
+           |  +--ro oc-transceiver:vendor-part?         string
+           |  +--ro oc-transceiver:serial-no?           string
+           |  +--ro oc-transceiver:date-code?           oc-yang:date-and-time
+           |  +--ro oc-transceiver:supply-voltage
+           |  |  +--ro oc-transceiver:instant?          decimal64
+           |  +--ro oc-transceiver:laser-bias-current
+           |     +--ro oc-transceiver:instant?          decimal64
+           +--rw oc-transceiver:physical-channels
+              +--rw oc-transceiver:channel* [index]
+                 +--ro oc-transceiver:state
+                    +--ro oc-transceiver:laser-temperature
+                    |  +--ro oc-transceiver:instant?    decimal64
+                    +--ro oc-transceiver:output-power
+                    |  +--ro oc-transceiver:avg?        decimal64
+                    +--ro oc-transceiver:input-power
+                    |  +--ro oc-transceiver:instant?    decimal64
+                    +--ro oc-transceiver:laser-bias-current
+                       +--ro oc-transceiver:instant?    decimal64
+```  
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Definition**                         |
+|--------------------------|-------------------------------------|
+| YANG                     | Yet Another Next Generation: modular language representing data structures in an XML tree format        |
+| gNMI                     | gRPC Network Management Interface: used to retrieve or manipulate the state of a device via telemetry or configuration data         |
+| XML                     | eXtensible Markup Language   |
+
+# 1 Feature Overview
+## 1.1 Requirements
+### 1.1.1 Functional Requirements
+1. Provide support for OpenConfig YANG models.
+2. Support telemetry for following read only parameters on transceiver components via REST and gNMI.
+3. Add support for following features:
+- serial-no
+- connector-type
+- vendor
+- vendor-part
+- date-code
+- supply-voltage
+- laser-temperature
+- output-power
+- input-power
+- laser-bias-current
+
```

Comment:
What about high/low warning/critical thresholds for temperature and voltage? These should also be encoded in transceiver DOM values.